docs/docsite/rst/user_guide/intro_inventory.rst

```diff
@@ -478,6 +478,96 @@ Here is an example of how to instantly deploy to created containers::
 .. note:: If you're reading the docs from the beginning, this may be the first example you've seen of an Ansible playbook. This is not an inventory file.
           Playbooks will be covered in great detail later in the docs.
 
+.. _using_multiple_inventory_sources:
+
+Using multiple inventory sources
+++++++++++++++++++++++++++++++++
+
+You can use multiple inventory sources if you want to target multiple environments, multiple inventory source types
+or reuse common groups and variables across multiple environments. The sources are merged and run just like
+only one inventory was given. One inventory can depend on groups or group variables of another inventory.
+Variable defined in the last inventory wins in precedence in accordance with :ref:`ansible_variable_precedence`.
+
+In the following example the inventory sources are directories but could be any of the supported inventory plugins
+or dynamic inventory scripts. In the example we have two different environments: production and staging. Both have some
+common variables and groups but may also have some own environment specific configurations.
+To avoid copy pasting the commmon groups and variables the following directory layout is created::
+
+    inventories/
+       production/
+          hosts               # inventory file for production servers
+          group_vars/
+             db-servers.yml   # assign variables to db servers in production environment
+       staging
+          hosts               # inventory file for staging environment
+       common/
+          hosts               # inventory file for defining common groups for all environments
+          group_vars/
+             all-servers.yml  # assign variables to all servers in all environments
+
+The production inventory is defined like this:
+
+.. code-block:: guess
+
+    [web-servers]
+    web1.example.com
+
+    [db-servers]
+    db1.example.com
+
+The staging environment is very similar:
+
+.. code-block:: guess
+
+    [web-servers]
+    web1.staging.example.com
+
+    [db-servers]
+    db1.staging.example.com
+
+The common inventory composes a new group named all-servers:
+
+.. code-block:: guess
+
+    [web-servers]
+
+    [db-servers]
+
+    [all-servers:children]
+    web-servers
+    db-servers
+
+The inventories are merged, so when you target common and production
+inventories all-servers group will have web1.example.com and db1.example.com as members and
+db1.example.com will have host variables from both db-servers.yml and all-servers.yml.
+In case of conflict a variable defined in db-servers.yml wins over all-servers.yml's variable.
+
+Without the empty web-servers and db-servers groups the common inventory would have a dependency on those
```

Comment:
Ah, this looks like it might define the use case. Can you add variables to the example, and then flesh this part out? Something like, "Without the empty web-servers and db-servers groups, the value of `my_var` would be X instead of Y" so it's easier to see the consequences (and advantages).